kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ClusterCondition.java

```diff
@@ -6,22 +6,34 @@
 
 package io.kroxylicious.kubernetes.operator;
 
+import java.util.Locale;
+import java.util.Optional;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.virtualkafkaclusterspec.TargetCluster;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
 import static io.kroxylicious.kubernetes.api.v1alpha1.kafkaproxystatus.clusters.Conditions.Status;
 
-public record ClusterCondition(String cluster, ConditionType type, Status status, String reason, String message) {
+public record ClusterCondition(@NonNull String cluster,
+                               @NonNull ConditionType type,
+                               @NonNull Status status,
+                               @Nullable String reason,
+                               @Nullable String message) {
 
     public static final String INVALID = "Invalid";
 
     static ClusterCondition accepted(String cluster) {
         return new ClusterCondition(cluster, ConditionType.Accepted, Status.TRUE, null, null);
     }
 
-    static ClusterCondition filterInvalid(String cluster, String filterName, String reason) {
+    static ClusterCondition filterInvalid(String cluster, String filterName, String detail) {
         return new ClusterCondition(cluster, ConditionType.Accepted, Status.FALSE, INVALID,
-                String.format("Filter \"%s\" is invalid: %s", filterName, reason));
+                String.format("Filter \"%s\" is invalid: %s", filterName, detail));
     }
 
-    static ClusterCondition filterNotExists(String cluster, String filterName) {
+    static ClusterCondition filterNotFound(String cluster, String filterName) {
         return new ClusterCondition(cluster, ConditionType.Accepted, Status.FALSE, INVALID,
                 String.format("Filter \"%s\" does not exist.", filterName));
```

Comment:
maybe these kinds of messages should have a namespace qualifier like `entity X not found in namespace Y`, since the named thing might exist but be in a mismatched namespace. (might be some IT coverage to add in this area now I think of it, checking that resources are isolated nicely by namespaces)

Reply:
https://github.com/kroxylicious/kroxylicious/issues/1911

Reply:
given we are no supporting cross namespace refs at the moment, and the message you are referring to gets written in the CR's status section, I don't think there is possibility to ambiguity right now.  Sure, if/when we allow refs to other namespaces, we should do what you say.